src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -394,6 +420,7 @@ def classpath_for_context(context):
     # all targets to finish.
     with self.invalidated(relevant_targets,
                           invalidate_dependents=True,
+                          invalidate_direct_checker=JvmCompile.enable_strict_deps,
```

Comment:
I am a little confused. What is not a global property? "JvmCompile.enable_strict_deps"? It is a static method thus I think as long as class JvmCompile is used, this method will work?